Picacomic: Fix json decoding error of non-ASCII chars (#19434)

fumiama opened the pull request:
Closes #19434
Caused by #19400

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [ ] Have tested the modifications by compiling and running the extension through Android Studio

I have no condition to test because the repo is too heavy, but it can work theoretically for its just a small change.

alessandrojean commented on src/zh/picacomic/src/eu/kanade/tachiyomi/extension/zh/picacomic/Picacomic.kt:
```diff
@@ -100,7 +100,7 @@ class Picacomic : HttpSource(), ConfigurableSource {
         }
 
-        val payload = parts[1]?.let { JSONObject(String(Base64.decode(it, Base64.DEFAULT))) }
+        val payload = parts[1]?.let { JSONObject(Base64.decode(it, Base64.DEFAULT).decodeToString()) }
```
While you're into it, if you can remove the `org.json` usages in favor of `kotlinx.serialization` (as the extension already uses in other places and even has a `Json` instance), it would be better. It's not a good practice to mix two parsers.

fumiama replied:
That's all right.

fumiama reviewed (commented):
Modified.

drelover commented:
I have updated the new version，but the  same problem still exist.

fumiama commented:
> I have updated the new version，but the  same problem still exist.

Yes, I have also updated and the same error occurs again. Maybe we should refer to some other parts. The newly introduced json lib may be a possible reason.
I am quite sure that this problem is caused by this line, because there is a symptom that when you reinstall it completely, the problem fixed for the first time and then occurred again.

alessandrojean commented:
>  The newly introduced json lib may be a possible reason

No as the line modified in particular doesn't even use kotlinx.serialization, it is still using the old `org.json.JSONObject`, which should be already replaced at this point.

fumiama commented:
> Please do follow the PR checklist next time and compile locally and test before submitting new untested PRs.

OK. Maybe it will cost me a few days or weeks to install all the environments so I will left this problem to someone else who would like to fix.
May my thoughts mentioned above help you.
